From 357d326f274c2a72569e0c453f29ce3233d739ae Mon Sep 17 00:00:00 2001
From: Alec Ari <neotheuser@ymail.com>
Date: Sun, 2 Jun 2019 20:00:29 -0500
Subject: arch/x86/Kconfig: Make SCHED_SMT optional again

Minimal runtime overhead is still overhead. This is for
a real-time kernel so everything counts.

Signed-off-by: Alec Ari <neotheuser@ymail.com>
---
 arch/x86/Kconfig | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index f1c002d13b2d..bd421ee17a6f 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -1020,7 +1020,13 @@ config NR_CPUS
 	  to the kernel image.
 
 config SCHED_SMT
-	def_bool y if SMP
+	bool "SMT (Hyperthreading) scheduler support"
+	depends on SMP
+	---help---
+	  SMT scheduler support improves the CPU scheduler's decision making
+	  when dealing with Intel Pentium 4 chips with HyperThreading at a
+	  cost of slightly increased overhead in some places. If unsure say
+	  N here.
 
 config SCHED_MC
 	def_bool y
-- 
2.20.1

